shared/js/background/declarative-net-request.js

```diff
@@ -63,16 +70,23 @@ function generateEtagRule (id, etag) {
 }
 
 /**
- * Find an existing dynamic declarativeNetRequest rule with the given rule ID
+ * Find an existing session or dynamic declarativeNetRequest rule with the given rule ID
  * and return it.
  * @param {number} desiredRuleId
  * @returns {Promise<import('@duckduckgo/ddg2dnr/lib/utils.js').DNRRule|null>}
  */
-async function findExistingDynamicRule (desiredRuleId) {
+async function findExistingSessionOrDynamicRule (desiredRuleId) {
```

Comment:
I think the code should be shared for fetching session/dynamic rules here, but I don't think the function should check both. I think the caller should specify which (session or dynamic) it wants to check.

My vote would be to have this function take another argument which is a bool to specify session/dynamic. Then to have two wrapper functions to make it nicer to use. Something like:

```javascript
async function findExistingSessionOrDynamicRule (session, desiredRuleId) {
    const rules = await chrome.declarativeNetRequest[session ? 'getSessionRules' : 'getDynamicRules']()
    ...
}

const findExistingDynamicRule = findExistingSessionOrDynamicRule.bind(null, false);
const findExistingSessionRule = findExistingSessionOrDynamicRule.bind(null, true);
```